parquet-column/src/main/java/org/apache/parquet/schema/Types.java

```diff
@@ -406,6 +408,11 @@ protected PrimitiveType build(String name) {
                   meta.getPrecision() <= MAX_PRECISION_INT64,
                   "INT64 cannot store " + meta.getPrecision() + " digits " +
                   "(max " + MAX_PRECISION_INT64 + ")");
+              if (meta.getPrecision() <= MAX_PRECISION_INT32) {
+                String msg = String.format("Consider use INT32 instead of INT64 for storing %d digits; see %s for details.",
+                    precision, "https://github.com/Parquet/parquet-format/blob/master/LogicalTypes.md");
```

Comment:
This URL should reference the Apache project, https://github.com/apache/parquet-format/blob/master/LogicalTypes.md and should probably be a static final constant.